### DISCUSSION TITLE
How to get keyboard inputs using the API?

### DISCUSSION THREAD

**Discussion by imanf94** [microsoft/AirSim#3816] [Support Q&A] [answered]
Hi, I'm using Airsim car client in a lane-keeping system project. I implemented a controller on the steering and throttle; however, I need to get some keyboard inputs from the user for lane changing.
I know there is a solution to get all the pressed keys in the python and process them as I want. But, I am looking for a cleaner solution to make sure that the pressed key is in the Airsim environment not anywhere in the windows.

**Comment by zimmy87** [comment, marked as answer]
You will probably have to do this via Unreal key bindings. You can take a look at ACarPawn::setupInputBindings() for an example of how the default controls for the car are configured.

**Reply by imanf94** [reply]
Thank you very much for your help.
For further references, these links may be helpful:
https://github.com/microsoft/AirSim/blob/master/Unreal/Plugins/AirSim/Source/Vehicles/Car/CarPawn.cpp
https://github.com/microsoft/AirSim/blob/master/Unreal/Plugins/AirSim/Source/AirBlueprintLib.cpp
